src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,7 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
```

Comment:
The header comment should be updated as well since you add the isMutating() checking in the method.